tools/project_manager/show_good_word_suggestions.php

```diff
@@ -35,15 +27,33 @@
 $format = get_enumerated_param($_REQUEST, 'format', 'html', array('html', 'file', 'update'));
 
 if($format=="update") {
-    $postedWords = parse_posted_words($_POST);
+    $is_reject = isset($_POST[REJECT_SUGGESTIONS]);
+    $destination_list = isset($_POST[BAD_WORD_LIST]) ? BAD_WORD_LIST : GOOD_WORD_LIST;
+
+    $postedWords = $is_reject ? array() : parse_posted_words($_POST);
 
-    $words = load_project_good_words($projectid);
-    $words = array_merge($words,$postedWords);
-    save_project_good_words($projectid,$words);
+    $words = $destination_list == GOOD_WORD_LIST ? load_project_good_words($projectid) : load_project_bad_words($projectid);
+    $words = array_merge($words, $postedWords);
+    if ($destination_list == GOOD_WORD_LIST)
+    {
+        save_project_good_words($projectid, $words);
+    }
+    else
+    {
+        save_project_bad_words($projectid, $words);
+    }
 
     $format="html";
 }
 
+$fileObject = get_project_word_file($projectid,"good");
+$timeCutoff = get_integer_param($_REQUEST, 'timeCutoff', $fileObject->mod_time, 0, null);
+
+if($timeCutoff==0)
+    $time_cutoff_text = _("<b>All proofreader suggestions</b> are included in the results.");
+else
+    $time_cutoff_text = sprintf(_("Only proofreader suggestions made <b>after %s</b> are included in the results."),strftime($datetime_format,$timeCutoff));
+
 list($all_suggestions_w_freq,$all_suggestions_w_occurrences,$round_suggestions_w_freq,$round_suggestions_w_occurrences,$rounds,$round_page_count,$messages) =
```

Comment:
Here's another fun edge case. If the user uses the new "add to bad list" function to add all suggestions to the bad words list, when the page loads it will say "There are no suggestions in the given time frame that aren't already on the Good Words List." -- which is 100% correct. However, the timestamp of the Good Words List file won't be reset, so the alert on the PM page will still show up. The ideal solution would be to have the alert on the PM listing use the same logic here to determine if there really are good words to show, but that's rather resource intensive. Instead, we could:
```
if($is_reject && $timeCutoff == $fileObject->mod_time && count($all_suggestions_w_occurrences) == 0)
{
    // touch good word list function call
}
```

But trying to implement that in the other file is going to be complicated, and really the whole thing is a mess (which is all my fault 13 years ago).

So for now let's not scope creep the hell out of this task and live with the problem for now.

Reply:
Wouldn't the touch get called during the is_reject update processing?

Reply:
Not in the case where they add the last word to the bad words list. It would if they rejected all the suggestions,

Reply:
I see, I was getting confused between "add to bad list" and reject all suggestions.